pages/positions/openloan.tsx

```diff
@@ -1,10 +1,295 @@
+  infoGroup: 'inline-flex w-full place-content-center pt-1',
+  loanInfoButton: 'bg-teal-900 rounded-2xl text-slate-200 text-[8px] font-semibold inline-flex mr-2 px-2 py-1 items-center',
+  infoIcon: 'mr-1',
+  sectionHeader: 'font-semibold text-gray-200 w-full',
+  collateralHeader: 'inline-flex w-full place-content-start',
+  collateralHeaderText: 'font-semibold text-gray-200',
+  selectCollateralButton: 'bg-[#274060] rounded-2xl text-slate-200 text-[8px] font-semibold inline-flex px-2 py-1 items-center mx-4',
+  chrevronIcon: 'ml-1',
+  confirmGrey: 'bg-[#274060] w-full rounded-2xl text-gray-500 inline-flex place-content-center py-2 font-semibold',
+  confirmInsuffBal: 'bg-red-400 w-full rounded-2xl text-slate-200 inline-flex place-content-center py-2 font-semibold',
+  confirmGreen: 'bg-green-300 w-full rounded-2xl text-slate-200 inline-flex place-content-center py-2 font-semibold',
+  confirmButtonContainer: 'pb-4 w-full',
+  interestRateText: 'w-full text-right text-gray-200 pr-4',
+  spacer: 'p-5',
+}
 
 const OpenLoanPage: NextPage = () => {
-    return (
-        <OpenLoanController />
-    )
+  const {openLoanHandler, token0, token1, setToken0, setToken1} = useOpenLoanHandler()
+  const [collateralType, setCollateralType] = useState<CollateralType>(CollateralType.None)
+  const [collateralButtonText, setCollateralButtonText] = useState<string>('Select collateral type')
+  const [confirmStyle, setConfirmStyle] = useState<string>(style.invalidatedButton)
+  const [loanAmt, setLoanAmt] = useState<number>(0)
+  const [loanAmtStr, setLoanAmtStr] = useState<string>('')
+  const [collateralAmt0, setCollateralAmt0] = useState<number>(0)
+  const [collateralAmt0Str, setCollateralAmt0Str] = useState<string>('')
+  const [collateralAmt1, setCollateralAmt1] = useState<number>(0)
+  const [collateralAmt1Str, setCollateralAmt1Str] = useState<string>('')
+  const {register, handleSubmit, setValue} = useForm()
+  const [isOpen, setIsOpen] = useState<boolean>(false)
+  const {notifyError, notifySuccess} = useNotification()
+  const [buttonText, setButtonText] = useState<string>('Confirm')
+  const [collateral1Class, setCollateral1Class] = useState<string>('')
+
+  useEffect(() => {
+    resetCollateralType()
+    validate()
+  }, [token0, token1])
+
+  useEffect(() => {
+    setIsOpen(false)
+    // console.log("selected collateral type", CollateralType[collateralType])
+    setCollateralButtonText(getCollateralTypeButtonText(collateralType))
+    setCollateralAmt1(0)
+    setCollateralAmt1Str('')
+    setCollateral1Class(collateralType == CollateralType.Both ? style.numberInputContainer : style.numberInputHidden)
+    validate()
+  }, [collateralType])
+
+  function getCollateralTypeButtonText(collateralType: CollateralType) {
+    switch (collateralType) {
+      case CollateralType.None:
+        return 'Select collateral type'
+      case CollateralType.LPToken:
```

Comment:
there are some functions that need to go to the custom hook, that's what it's used for. If it gets too big, you can also separate them into smaller, more atomic custom hooks.

Reply:
Done